### PR TITLE
[WIP]AC: support LM 1B

### PR DESCRIPTION
**Pull request by eaidova**
- [x] dataset converter
- [x] representations
- [x] LSTM input
- [x] seq inference
- [x] adapter
- [x] multi inference input filling
- [x] metric
- [ ] readme

### CODE REVIEW

**Comment by eaidova** [comment]
Jenkins please retry a build

File: models/public/language-model-0001/language-model-0001.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# language-model-0001
```
Comment (eaidova) [inline models/public/language-model-0001/language-model-0001.md:1]: for public models we prefer to save original name,  please return it

File: models/public/language-model-0001/language-model-0001.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+## Specification
+
+| Metric            | Value              |
+|-------------------|--------------------|
+| Type              | Language Prediction|
+| Source framework  | TensorFlow\*       |
+
+## Accuracy
+
+Test on the test dataset `language-model-0001-dataset`, At convergence, the perplexity should be around 30.
```
Comment (eaidova) [inline models/public/language-model-0001/language-model-0001.md:24]: please give more specific number, just `around 30` not enough for understanding accuracy drops in validation cases

File: models/public/language-model-0001/language-model-0001.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  --input_data: Input data files for eval model.
+  --max_eval_steps: Maximum mumber of steps to run "eval" mode.
+  --max_sample_words: Sampling stops either when </S> is met or this number of steps has passed.
+  --mode: One of [sample, eval, dump_emb, dump_lstm_emb]. "sample" mode samples future word predictions, using FLAGS.prefix as prefix (prefix could be left empty). "eval" mode calculates perplexity of the FLAGS.input_data. "dump_emb" mode dumps word and softmax embeddings to FLAGS.save_dir. embeddings are dumped in the same order as words in vocabulary. All words in vocabulary are dumped.dump_lstm_emb dumps lstm embeddings of FLAGS.sentence to FLAGS.save_dir.
+  --num_samples: Number of samples to generate for the prefix.
+  --pbtxt: GraphDef proto text file used to construct model structure.
+  --prefix: Used for "sample" mode to predict next words.
+  --save_dir: Used for "dump_emb" mode to save word embeddings.
+  --sentence: Used as input for "dump_lstm_emb" mode.
+  --vocab_file: Vocabulary file.
```
Comment (eaidova) [inline models/public/language-model-0001/language-model-0001.md:43]: this section should contains info about input layer of original model (layer name, its shape, layout and info which input data expected

File: models/public/language-model-0001/language-model-0001.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+### Converted model
+
+- `i` - The prefix used for word predictions
```
Comment (eaidova) [inline models/public/language-model-0001/language-model-0001.md:47]: also have nothing common with input layers description

File: models/public/language-model-0001/language-model-0001.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+## Output
+
+### Original model
+
+In the sample mode, the model output a prediction word.
+In the eval mode, the model output the perplexity of each sentence.
+
+### Converted model
+
+The output of converted model is similar to the one of original model.
```
Comment (eaidova) [inline models/public/language-model-0001/language-model-0001.md:59]: which exactly? Does it mean that converted model  can return perplexity? This description should correspond with conversion parameters in model.yml

File: models/public/language-model-0001/model.yml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    size: 409600228
+    sha256: a96f0e035ba6901812e55dca415b3b40949372e9dd6e009f04da56816eb594d3
+    source: http://download.tensorflow.org/models/LM_LSTM_CNN/all_shards-2016-09-10/ckpt-softmax7
+  - name: language-model-0001-ckpt-softmax8
+    size: 3200206
+    sha256: fd65afd5520c054d74b16570f2bfb8454bc1296d5cfb816d70659e61664b6cb1
+    source: http://download.tensorflow.org/models/LM_LSTM_CNN/all_shards-2016-09-10/ckpt-softmax8
+model_optimizer_args:
+  - --input_model=$dl_dir/language-model-0001.pbtxt
+  - --input_checkpoint=$dl_dir_ckpt
```
Comment (eaidova) [inline models/public/language-model-0001/model.yml:94]: here no such directory

File: tools/accuracy_checker/accuracy_checker/launcher/input_feeder.py [hunk 7]
```diff
@@ -122,7 +124,7 @@ def fill_non_constant_inputs(self, data_representation_batch):
                 identifiers = data_representation.identifier
                 data = data_representation.data
                 if not isinstance(identifiers, list) and input_regex is None:
-                    input_data = data
+                    input_data = data if num_of_non_constant_inputs <= 1 else data[input_layer_num]
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/launcher/input_feeder.py:127]: @liyong4x  could you please describe the reason of these changes? It breaks general pipeline, so we need to find another solution

**Comment by eaidova** [comment]
Jenkins please retry a build

**Comment by IRDonch** [comment]
@eaidova This has been WIP since January, maybe it's time to close it?

**Pull request closed without merging**